### PR TITLE
add basic SyncTeX support

### PR DESCRIPTION
**Pull request by andrewrynhard**

### CODE REVIEW

**Comment by andrewrynhard** [comment]
@mungerd: I would love it if we could merge this! I know you have put the work in so I would like your permission first.

**Comment by andrewrynhard** [comment]
Hmm. I don't know what was changed but this is no longer working.

**Comment by mungerd** [comment]
@andrewrynhard: I didn't make a pull request because there are still two things that I don't understand: (1) it doesn't work if the `output` directory starts with a `.` as `.latex`; (2) syncing the PDF view from the LaTeX document works when the PDF is already open and up-to-date, but sometimes doesn't work if the PDF needs to be regenerated (some delays somewhere, maybe).

I'd be happy to fix it if you could give me some pointers.

And of course, you are welcome to integrate the patch!

**Comment by andrewrynhard** [comment]
@mungerd: Interesting. I will have some time later this week to take a good look at it. Its interesting because before the synctex code was moved into its own file, it was working fine on my machine. Now I can't get it to work at all. I will run through the diff. Thanks for getting started on this!

**Comment by mungerd** [comment]
@andrewrynhard: Okay, let me know if I can help.  Here it works with both Atom 1.1.0 and 1.2.3, under Linux.

**Comment by mungerd** [comment]
@andrewrynhard: Are you getting any error message or it just does nothing?

**Comment by andrewrynhard** [comment]
@mungerd: It seems to be working when I click on the PDF only. I remember a bidirectional syncing. Is this intentional or is this what you mean when you said "syncing the PDF view from the LaTeX document works when the PDF is already open and up-to-date..."? Also, I have my output going to '.latex' and it seems to be working. I think there are still some small issues with latex-plus outside of the synctex code. By the way I am on OS X. I can run a linux machine later this week to see if I get the same issues you mentioned.

**Comment by mungerd** [comment]
@andrewrynhard: The syncing from PDF view to LaTeX source is handled by pdf-view (perhaps through pdf.js), I did not do this myself.  And this works very well, in all instances, regardless of the `output` directory.

What I have added is syncing from LaTeX source to PDF view: **hitting `ctrl-;` scrolls the PDF view** to the line that contains the text that is under the cursor in the editor.

But, in my Linux environment:
- hitting `ctrl-;` _never_ scrolls the PDF view if the `output` directory starts with a `.` as `.latex`;
- hitting `ctrl-;` _sometimes_ fails to scroll the PDF view to the right line if the document needs to be recompiled firsthand.
  Other than that, it works.  It's good enough for my own needs, but I would love to make it convenient enough for other users.

**Comment by andrewrynhard** [comment]
@mungerd: ok now I understand. Thank you. I have a copy of the synctex code before you moved it to its own file. I will have to take a look and see what is different. I have been in the middle of a move over the last couple weeks and should be done with that by this weekend. The earliest I can take a look is next week! We will get this working. I am sure that the users of latex-plus will appreciate it.

**Comment by mungerd** [comment]
@andrewrynhard: Thanks for the update!  Please, take your time.  I myself have very little spare time currently.

File: lib/compiler.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ export default class Compiler {
       args.push("-bibtex");
     }
 
+    if (atom.config.get("latex-plus.synctexEnabled")) {
+      args.push("-synctex=1");
+    }
+
```
Comment (dpo) [inline lib/compiler.js:53]: I don't see any reason to ever disable syncTeX, so perhaps such a setting isn't necessary.

**Comment by andrewrynhard** [comment]
@dpo once this is merged, synctex will always be enabled. I have merged this locally and working on a fix now. Hopefully I can have it fixed by this weekend.

**Comment by andrewrynhard** [comment]
@mungerd do you think that a mouse click event should invoke synctex instead of `cmd-;`?

**Comment by dpo** [comment]
A mouse click already works for reverse sync (if I create a symlink to the `synctex.gz` file in the working folder). I think that's great. Forward sync should be bound to a key combination, I suppose. But `cmd-;` is already bound though.

**Comment by andrewrynhard** [comment]
@dpo should we want to update the PDF before syncing in case there are any changes?

**Comment by dpo** [comment]
That would be a user error, IMO. On a small document, it's inconsequential, but if I'm working on an entire book, I'd rather not trigger recompilation if I forward sync (by mistake or otherwise). The worst that will happen if you don't recompile is that sync will take you to the wrong place.

**Comment by andrewrynhard** [comment]
@dpo that makes sense. I will bind it to a different key.

**Comment by dpo** [comment]
Just in case this is relevant: https://github.com/izuzak/atom-pdf-view/pull/111

**Comment by mungerd** [comment]
Thanks guys for making this work.  Sorry I wasn't able to answer earlier.

**Comment by andrewrynhard** [comment]
@mungerd not a problem! Thanks for your work on this!

**Pull request merged**